play.pokemonshowdown.com/src/panel-chat.tsx

```diff
@@ -546,12 +559,28 @@ export class ChatTextEntry extends preact.Component<{
 	miniedit: MiniEdit | null = null;
 	history: string[] = [];
 	historyIndex = 0;
+	tabComplete: {
+		candidates: string[][],
+		index: number,
+		prefix: string,
+		cursor: string | number | null,
+		reset: () => void,
+	} = {
+			candidates: [],
+			index: 0,
+			prefix: '',
+			cursor: null,
+			reset() {
+				this.cursor = null;
+			},
+		};
 	override componentDidMount() {
 		this.subscription = PS.user.subscribe(() => {
 			this.forceUpdate();
 		});
-		const textbox = this.base!.children[0].children[1] as HTMLElement;
-		if (textbox.tagName === 'TEXTAREA') this.textbox = textbox as HTMLTextAreaElement;
+		const textbox = (this.base!.children[0].children[1] ||
+			this.base!.querySelector('textarea[name=textbox]')) as HTMLElement;
+		if (textbox.tagName === 'TEXTAREA' || textbox.tagName === 'PRE') this.textbox = textbox as HTMLTextAreaElement;
```

Comment:
Oh nooooo. If it's a `pre` you shouldn't be manipulating its contents directly. It's tracked in `room.miniedit`. See how `toggleFormatChar` does it.